### PR TITLE
23604 update logic for changing business name

### PR DESCRIPTION
**Pull request by meawong**
*Issue #:* /bcgov/entity#23604

*Description of changes:*
- update logic for changing business name

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-599-h6gsnubn.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-599-h6gsnubn.web.app

File: src/components/common/YourCompany/ChangeBusinessType.vue [hunk 1]
```diff
@@ -433,24 +433,20 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
   submitTypeChange () {
     this.setEntityType(this.selectedEntityType)
     this.isEditingType = false
-    if (this.shouldUpdateName()) {
```
Comment (meawong) [inline src/components/common/YourCompany/ChangeBusinessType.vue:436]: We would only update the name and set the NR if the business name had **NOT** changed. This prevents the case when a user attempts to change the business type (and therefore the name) more than once, as the business name **HAS** changed.

File: src/components/common/YourCompany/ChangeBusinessType.vue [hunk 2]
```diff
@@ -433,24 +433,20 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
...
-        } as any)
-        this.setNameRequestLegalName(updatedName)
         this.setNameChangedByType(true)
       }
     }
   }
 
-  shouldUpdateName (): boolean {
-    return this.isNumberedCompany && !this.hasBusinessNameChanged
-  }
```
Comment (meawong) [inline src/components/common/YourCompany/ChangeBusinessType.vue:452]:
```
/** Whether business name has changed. */
    hasBusinessNameChanged (): boolean {
      const currentLegalName = this.getNameRequestLegalName // may be empty
      const originalLegalName = this.getOriginalLegalName // from original business record
      return (currentLegalName !== originalLegalName)
}
    
```

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-599-h6gsnubn.web.app

File: src/views/Alteration.vue [hunk 3]
```diff
@@ -214,6 +214,7 @@ export default class Alteration extends Mixins(CommonMixin, FeeMixin, FilingTemp
     }
   }
 
+  @Watch('hasBusinessNameChanged')
```
Comment (severinbeauvais) [inline src/views/Alteration.vue:217]: I don't know if this change is the reason why, but I found that I can Undo the Company Name and the Business Type remains changed.

image

Should this make the filing invalid?

Should the Business Type also be undone?

**Review by severinbeauvais — COMMENTED** [review]
Good analysis and I agree with the solution, but please test a bit more (including saving and resuming) to make sure all edge cases are OK, too. Thanks!

File: src/components/common/YourCompany/ChangeBusinessType.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/common/YourCompany/ChangeBusinessType.vue:436]: Does this logic work on a named company? With and without a new NR?

File: src/views/Alteration.vue (hunk 3 above)
Comment (severinbeauvais) [reply src/views/Alteration.vue:217]: Oh, wait.

When I saved and resumed, the resume was in the old Edit UI. Let me test this again.

**Review by severinbeauvais — APPROVED** [review]

File: src/views/Alteration.vue (hunk 3 above)
Comment (meawong) [reply src/views/Alteration.vue:217]: I just checked this on dev and this issue already exists but I can look into fixing this. Should I update the scope on this ticket?

Comment (severinbeauvais) [reply src/views/Alteration.vue:217]: Please test this thoroughly. I believe it's working well.

**Comment by ArwenQin** [comment]
I did some tests, seems good!

**Review by ArwenQin — APPROVED** [review]

File: src/components/common/YourCompany/ChangeBusinessType.vue (hunk 1 above)
Comment (meawong) [reply src/components/common/YourCompany/ChangeBusinessType.vue:436]: Since these changes still retain the check for a numbered business, as before, named companies should not be reaching this block of code below. There should be separate logic that handles named businesses (I can verify where this is if you like).

Comment (meawong) [reply src/components/common/YourCompany/ChangeBusinessType.vue:436]: > I did a little test on a named BC Limited, and could not change the type to ULC without a NR.

I thought this was the expected behaviour for named businesses? Since altering the business type for a named business would require a name change and this requires an NR.

Comment (severinbeauvais) [reply src/components/common/YourCompany/ChangeBusinessType.vue:436]: Yes, but what if wanted to change my named BC to a numbered ULC? Anyway, we discussed this previously and decided it was low priority right now.

File: src/components/common/YourCompany/ChangeBusinessType.vue [hunk 4]
```diff
@@ -433,24 +433,20 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
   submitTypeChange () {
     this.setEntityType(this.selectedEntityType)
     this.isEditingType = false
-    if (this.shouldUpdateName()) {
+    if (this.isNumberedCompany) {
       const originalName = this.getOriginalLegalName
       const updatedName = this.getUpdatedName(originalName)
+      this.setNameRequest({
+        legalType: this.selectedEntityType,
+        nrNum: this.getOriginalNrNumber
```
[the remark this answers is not in the document]
Comment (meawong) [reply src/components/common/YourCompany/ChangeBusinessType.vue:441]: The original NR is an empty string for a numbered company.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-599-h6gsnubn.web.app

**Review by severinbeauvais — APPROVED** [review]
I did some quick tests and it LGTM!

Please list out, in the ticket, the tests you did. Great job! Thanks!
